content/knowledge_base/flare/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,210 @@
+Avalanche treats the concept of repeated queries and multiple counters from Snowball taking advantage of DAG structure.
+Specifically, when a transaction `T` is queried, all of its ancestors reachable through the DAG edges from it are implicitly part of the query.
+Consequently, nodes would respond positively to a query for transaction `T` only if `T` and all of its ancestors are the preferred option in their respective conflict sets.
+If more than a given threshold of responders vote positively, the transactions gets a [chit](../../../glossary.md#Chit).
+After that nodes compute their level of confidence as the sum of transaction chit and the chits its progeny, meaning they query the transaction just once and rely on new vertices and possible chits added to the progeny to build up their confidence.
+
+Described ties are broken in case of preference for first-seen transactions. TODO: explain this
+Chits can also be decoupled from the DAG structure, making the protocol immune to attacks where the attacker generates large padded subgraphs.
+
+## Consensus Specification
+
+Each node `u` keeps track of all transactions it has learned about in set Tu. The set Tu is [partitioned](../../../glossary.md#Partition-of-a-Set) into mutually exclusive conflict sets - PT (partition of 'T'), and PT ∈ Tu.
+Since conflicts are [transitive](../../../glossary.md#Transitive-Relation), if PTi and PTj are conflicting they belong to the same conflict set of transactions — PTi = PTj.
+Conflicting transation have the equivalence relation because they are equivocations spending the same UTXO.
+
+The parent connection to T' from T is expressed as `T' <— T`.
+The relation `T' *<— T` is its reflexive [transitive closure](../../../glossary.md#Transitive-Closuse-of-a-Graph) meaning that there is path from T to T'.
+If T' <— T, then every node in the system that has T will also have T' and will know about the same relation `T' <— T`.
+If such a relation between these transactions do not exist, then no node will end up with T' <— T.
+
+DAGs built in different nodes are guaranteed to be compatible, though at any moment in time they might not have a complete view of all vertices in the system.
+Each node can compute a confidence value `d U (T)` (confidence value for node `u` about transaction `T`) from the progeny as the sum of transaction T decendants cihts from the progeny and its own chit.
+Each transaction initially has a chit of 0 before the node gets the query results.
+If the node collects a threshold of α 'yes' votes from the query the value of transaction T chit is set to 1, otherwise it will stay foerever 0.
+A chit value reflect the result from a single query of its associated transaction and becomes immutable afterwards, while confidence `d` can increase as the DAG grows by collecting more chits in its progeny.
+Additionally each node maintains its own local list of known nodes: Nu — nodes known to `u` is a subset of the set of all existing nodes: `Nu ⊆ N`.
```

Comment:
```suggestion
Additionally, each node maintains its own local list of known nodes: Nu — nodes known to `u` is a subset of the set of all existing nodes: `Nu ⊆ N`.
```

Reply:
Added the comma here https://github.com/optakt/optakt.github.io/pull/13/commits/e0e27e6be8c0aef4153af5e2add9722380c23a59#diff-2d4f16e70781857aec6689a748ad155e45611b81f184aee076bb139890410b98R63
Thanks!